packages/odyssey-react/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Children, isValidElement, cloneElement, useMemo, CSSProperties, ComponentPropsWithRef, forwardRef } from "react";
+import type { ReactNode } from "react";
+import { nanoid } from 'nanoid';
+
+import styles from "./Icon.module.scss";
+
+export type Props = {
+  /**
+   * Title text used by screen readers
+   * @default the name of the icon
+   */
+  title?: string;
+
+  /**
+   * Id used to link title and svg elements, used by screen readers
+   * @default a randomly generated id
+   */
+  titleId?: string;
+
+  /**
+   * Size in any valid css unit, `px`, `em`, `%`, etc...
+   * @default 1em
+   */
+  size?: string;
+
+  /**
+   * Color of the icon fill in hex format, eg: #000000
+   * @default current text color
+   */
+  color?: string;
+} & ComponentPropsWithRef<'svg'>;
+
+/** 
+ * A system of icons which establishes a visual language
+ * that can be easily understood regardless of age, language or culture.
+ */
+
+const Icon = forwardRef<SVGSVGElement, Props>((
+  {
+    title,
+    titleId,
+    size,
+    color,
+    children
+  }, 
+  ref 
+) => {
+  
+  const memoId = useMemo(() => ('icon_'+nanoid(6)), []);
+
+  if(!titleId){
+    titleId = memoId
+  }
+
+  const sizeAndColor:CSSProperties = new Object();
+
+  if (size) {
+    sizeAndColor.fontSize = size;
+  }
+
+  if (color) {
+    sizeAndColor.color = color;
+  }
+
+  return (
+    <>
+      { 
+        Children.map<ReactNode, ReactNode>(children, child => {
+          if(isValidElement(child)) {
+            return cloneElement(
+              child, 
+              {
+                "aria-labelledby": titleId,
+                className: styles.root,
+                style: sizeAndColor,
```

Comment:
inline styles for this? Can we do this via class names instead? That way we have a nice hook for theming in the future rather than having to rework this logic

Reply:
Yes, this inline style is only ever created by passing props.

To achieve the same thing with classes would require that we have a finite set of specific sizes and colors for icons, which I do not believe we have be able to agree on.

Reply:
Do we need these overrides available at present? Designers have limited ability to modify either value today.

Icon isn't likely to be used by itself, so `currentColor` and scaling to a parent may continue to be sufficient for awhile. 

When we reach a breaking point, maybe we can sync with @joejezowski-okta to see which color applications we should support. (e.g. `.primaryStroke`, `.dangerStroke`)

Reply:
I'm +1 on a finite set of sizes and colors for now, in fact the colors should probably just be exposed via another level of abstraction so the API looks like this:

`<Icon color="error" size="large" />`

color would map to specific theme variable colors for now. This way we can use all classes and allow values to be themed as well.

Reply:
oh and and default like `fill: 'currentColor';` would make sense as well!